MetaMask/metamask-extension
# Handle extension unloaded and reloaded error: "Extension context invalidated" during dapp use for chromium browsers

## Explanation

If the extension is loaded or reloaded during a user session, the dapp will break, and when attempting to use the dapp, the error, `'Extension context invalidated.', will show in the console for chromium browsers. This PR provides a slightly more helpful error message by prompting the user to refresh the page.

This error occurs for both MV2 and MV3 builds.

Note: We are not yet handling this error on Firefox. This will be handled along with this issue - https://github.com/MetaMask/metamask-extension/issues/16909

## More Information

Fixes: #15975

## Screenshots/Screencaps



### Before
<img width="460" alt="Screenshot 2022-11-01 at 2 22 49 AM" src="https://user-images.githubusercontent.com/20778143/199092321-6956dd36-6bb6-4e36-b8b6-46eb143b9cb0.png">

### After

<img width="586" alt="Screenshot 2022-10-29 at 1 22 21 AM" src="https://user-images.githubusercontent.com/20778143/198708476-e66cb1db-156f-410d-b54d-bbf9522279f6.png">

## Manual Testing Steps

- Start the extension using MV3 build (yarn start:mv3)
- View a dapp page
- Notice the dapp is working
- Go to chrome://extensions/
- Turn off the extension, or toggle the extension on and off
- Attempt to use the dapp and observe the error in the console

## Pre-Merge Checklist


### + If there are functional changes:
